django/db/models/sql/datastructures.py

```diff
@@ -146,3 +161,9 @@ def as_sql(self, compiler, connection):
 
     def relabeled_clone(self, change_map):
         return self.__class__(self.table_name, change_map.get(self.table_alias, self.table_alias))
+
+    def equals(self, other, with_filtered_relation):
+        if isinstance(self, other.__class__):
```

Comment:
I think this condition could be the first line in the condition, and then you can scratch the last `return`.

Reply:
exactly, as I wrote above for the other `__eq__`

```py
return (
    isinstance(self, other.__class__) and
   self.table_name == other.table_name and
   self.table_alias == other.table_alias
)
```